cli/cmd/stat_summary.go

```diff
@@ -0,0 +1,224 @@
+	}
+
+	resp, err := client.StatSummary(context.Background(), req)
+	if err != nil {
+		return "", fmt.Errorf("error calling stat with request: %v", err)
+	}
+
+	return renderStatSummary(resp), nil
+}
+
+func renderStatSummary(resp *pb.StatSummaryResponse) string {
+	var buffer bytes.Buffer
+	w := tabwriter.NewWriter(&buffer, 0, 0, padding, ' ', tabwriter.AlignRight)
+
+	writeStatTableToBuffer(resp, w)
+	w.Flush()
+
+	// strip left padding on the first column
+	out := string(buffer.Bytes()[padding:])
+	out = strings.Replace(out, "\n"+strings.Repeat(" ", padding), "\n", -1)
+
+	return out
+}
+
+type summaryRow struct {
+	meshed      string
+	requestRate float64
+	successRate float64
+	latencyP50  int64
+	latencyP99  int64
+}
+
+func writeStatTableToBuffer(resp *pb.StatSummaryResponse, w *tabwriter.Writer) {
+	nameHeader := "NAME"
+	maxNameLength := len(nameHeader)
+
+	stats := make(map[string]*summaryRow)
+
+	for _, statTable := range resp.GetOk().StatTables {
+		table := statTable.GetPodGroup()
+		for _, r := range table.Rows {
+			name := r.Resource.Name
+			if name == "" {
+				continue
+			}
+
+			if len(name) > maxNameLength {
+				maxNameLength = len(name)
+			}
+
+			stats[name] = &summaryRow{
+				meshed:      strconv.FormatUint(r.MeshedPodCount, 10) + "/" + strconv.FormatUint(r.TotalPodCount, 10),
+				requestRate: getRequestRate(*r),
+				successRate: getSuccessRate(*r),
+			}
+		}
+	}
+
+	fmt.Fprintln(w, strings.Join([]string{
+		nameHeader + strings.Repeat(" ", maxNameLength-len(nameHeader)),
+		"MESHED",
+		"IN_RPS",
+		"IN_SUCCESS",
+		"IN_LATENCY_P50",
+		"IN_LATENCY_P99\t", // trailing \t is required to format last column
+	}, "\t"))
+
+	sortedNames := sortStatSummaryKeys(stats)
+	for _, name := range sortedNames {
+		fmt.Fprintf(
+			w,
+			"%s\t%s\t%.1frps\t%.2f%%\t%dms\t%dms\t\n",
+			name+strings.Repeat(" ", maxNameLength-len(name)),
+			stats[name].meshed,
+			stats[name].requestRate,
+			stats[name].successRate*100,
+			stats[name].latencyP50,
+			stats[name].latencyP99,
+		)
+	}
+}
+
+func buildStatSummaryRequest() (*pb.StatSummaryRequest, error) {
+	var requestParams util.StatSummaryRequestParams
+	requestParams.TimeWindow = timeWindow
+	requestParams.ResourceName = resourceName
+	requestParams.ResourceType = resourceType
+	requestParams.Namespace = namespace
+	requestParams.OutToName = outToName
+	requestParams.OutToType = outToType
+	requestParams.OutToNamespace = outToNamespace
+	requestParams.OutFromName = outFromName
+	requestParams.OutFromType = outFromType
+	requestParams.OutFromNamespace = outFromNamespace
+
+	return util.BuildStatSummaryRequest(requestParams)
+}
+
+func getRequestRate(r pb.StatTable_PodGroup_Row) float64 {
+	if r.Stats == nil {
+		return 0.0
+	}
+	success := r.Stats.SuccessCount
+	failure := r.Stats.FailureCount
+	window, err := util.GetWindowString(r.TimeWindow)
+	if err != nil {
+		fmt.Println(err.Error())
+	}
```

Comment:
I think this error check should also include an early return. Otherwise the subsequent call the `ParseDuration` is definitely going to fail. I'm actually not sure it make sense to print the error message at all, since that will disrupt the stat output. You could `log.Error(err.Error())` though, and that would be printed when the `--verbose` flag is used.